src/test/skript/tests/bukkit/potion module.sk

```diff
@@ -0,0 +1,620 @@
+		apply {_stronger} to {_entity}
+		clear the active and hidden speed effects of {_entity}
+		assert the active and hidden speed effects of {_entity} are not set with "holder still has speed effects"
+
+	# test adding/removing
+	apply {_stronger} to {_entity}
+	assert compare_timespans(30 seconds, the duration of {_entity}'s speed effect) is true with "duration is not 30 seconds"
+	add 15 seconds to {_entity}'s speed effect
+	assert compare_timespans(45 seconds, the duration of {_entity}'s speed effect) is true with "duration is not 45 seconds"
+	remove 30 seconds from {_entity}'s speed effect
+	assert compare_timespans(15 seconds, the duration of {_entity}'s speed effect) is true with "duration is not 15 seconds"
+	clear the speed effects of {_entity}
+
+	# test multiple hidden effects
+	parse if running minecraft "1.20.4":
+		apply {_weaker} to {_entity}
+		apply {_stronger} to {_entity}
+		apply {_strongest} to {_entity}
+		assert the active speed effect of {_entity} is {_strongest} with "holder doesn't have active speed effect"
+		assert the hidden speed effects of {_entity} are {_stronger} and {_weaker} with "holder doesn't have hidden speed effects"
+		clear the active speed effect of {_entity}
+		assert the active speed effect of {_entity} is {_stronger} with "holder doesn't have active speed effect"
+		assert the hidden speed effects of {_entity} are {_weaker} with "holder doesn't have hidden speed effects"
+		clear the speed effects of {_entity}
+		# test specific removals
+		apply {_weaker} to {_entity}
+		apply {_stronger} to {_entity}
+		apply {_strongest} to {_entity}
+		remove ambient potion effect of speed from {_entity}'s active and hidden speed effects
+		assert the speed effects of {_entity} are {_stronger} with "holder doesn't have stronger speed effect"
+		assert the active speed effect of {_entity} is {_stronger} with "holder doesn't have active stronger speed effect"
+		assert the hidden speed effects of {_entity} are not set with "holder has hidden speed effects"
+
+		# test invalid usage
+		parse:
+			clear the hidden speed effects of {_entity}'s tool
+		assert last parse logs is "Items (such as potions or stews) do not have hidden effects" with "hidden speed effects of item did not error"
+
+	# cleanup
+	delete the entity within {_entity}
+
+# EffApplyPotionEffect
+test "potion effect application":
+	# setup
+	spawn a pig at event-location:
+		set {_entity} to event-entity
+
+	# test simple application
+	apply speed to {_entity}
+	assert {_entity} has potion effect of speed
+	clear potion effects of {_entity}
+
+	# test timed application
+	apply slowness to {_entity} for 13 seconds
+	assert {_entity} has potion effect of slowness
+	assert compare_timespans(13 seconds, the duration of the first element of the potion effects of {_entity}) is true with "duration is not 13 seconds"
+
+	# cleanup
+	delete the entity within {_entity}
+
+# ExprPotionAmplifier
+test "potion amplifier property":
```

Comment:
this in specific, but everything should have tests for its other pattern variations
I believe you'd have caught the `level` conflict if so